_posts/2017-01-16-Jak-psat-php-Uloziste-kodu+databaze+frameworky.md

```diff
@@ -0,0 +1,46 @@
+
+[Bitbucket](https://bitbucket.org/) není pro projekty tak rozšířený jako předchozí varianty, jako jediný ale nabízí podporu [verzovacího systému Mercurial](https://www.mercurial-scm.org/). Navíc velmi ochotně spolupracuje s dalšími aplikacemi od firmy [Atlassian](https://www.atlassian.com/) jako je [Confluence](https://www.atlassian.com/software/confluence) nebo [Jira](https://www.atlassian.com/software/jira). 
+
+Mezi dalšími chci zmínit [CodePlex](https://www.codeplex.com/) Mezi služby, které nechi doporučovat patří [SourceForge](https://sourceforge.net/), který měl veliké problémy s přibalováním nevyžádaných probramů do všech projektů. Používal se i [Google Code](https://code.google.com/archive/), který ale v roce 2016 ukončil svou činnost a je dostupný jen ve formě archívu. 
+
+# Databáze
+Databáze jsou nedílnou součástí PHP i webových aplikací obecně, přesto je vezmu velmi zkrátka - je to disciplína hodná vlastního blogu (šance založit jakpsatdb.cz!). 
+
+Databáze slouží k ukládání dat. Data jdou řešit i jinak, například ukládat do textových souborů či posílat někam mimo aplikaci. Výhodou databází je ale přístupová rychlost a snadná udržitelnost a zálohovatelnost. 
+
+Databáze se obecně dělí na strukturované (relační) a nestukturované. Relační databáze mají pevně danou strukturu tabulek a sloupců v nich. Asi nejpoužívanější je [MySQL](https://www.mysql.com/) od firmy Sun. Další známé systémy jsou [MariaDB](https://mariadb.org/) jako opensource klon MySQL, oblíbený [PostgresQL](https://www.postgresql.org/), [Firebird](http://www.firebirdsql.org/) používaný hlavně v komerční sféře a jednoduchý [SQLite](https://sqlite.org/) vhodný pro malé projekty. 
+
+Do nestukturovaných databází (známých i jako noSQL) se většinou informaci ukládají ve tvaru `klíč:hodnota`. Používají se zejména tam, kde by byla nejasná či příliš složitá struktua relační databáze či při požadavcích na velkou rychlost a objem v jednoduché struktuře, jako je tomu například u dat generovaných z Internet of Things (internetu věcí). Mezi používané NoSQL systémy se řadí [MongoDB](https://www.mongodb.com/) nebo [Redis](https://redis.io/). 
+
+Z pohledu ovládání databáze můžete používat funkce přímo z PHP, jako třeba [mysqli](https://secure.php.net/manual/en/book.mysqli.php) nebo [PDO](https://secure.php.net/manual/en/book.pdo.php). Často se také používá nějaká externí knihovna. Knihovny můžou pracovat s jazykem SQL napřímo, jako kupříkladu [Dibi](https://www.dibiphp.com/), nebo pomocí objektově relačního mapování (ORM), jako třeba [Doctrine](http://www.doctrine-project.org/). 
+
+Pro ovládání databáze uživatelem se nejvíce používá [phpMyAdmin](https://www.phpmyadmin.net/). Za sebe můžu doporučit mnohem lehčí [adminer](https://www.adminer.org/cs/) od [Jakuba Vrány](https://www.vrana.cz/). 
```

Comment:
To jsme dva, ale někde je self-uploaded adminer blokovaný s důvodem "nemůžeme to kontrolovat, takže to určitě bude outdated a tím se tam vytvoří díra". Bohužel. Adminer už doporučuji jako lepší volbu, ještě jsem drobet upravil, aby to vyznělo jasněji